### DISCUSSION TITLE
TypeScript support

### DISCUSSION THREAD

**Discussion by Fi1osof** [suren-atoyan/monaco-loader#9] [Ideas]
Hi there!

Thanks for awesome solution! Without this i can not integrate monaco-editor with next.js due manaco-editor's styles in node_modules. (withCSS is deprecated). But will be great if add TypeScript support. For monaco instance i do this:

```tsx
import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api'

// ....

loader
  .init()
  .then((monaco: typeof monacoEditor) => {
    console.log('monaco', monaco)

    monaco.editor.create(editorContainer, {
      value: '// some comment',
      language: 'javascript',
    })
  })
```

But which props i can pass into loader.config({...});?

Maybe it's not necessary cause required props resolves via monaco: typeof monacoEditor.
```
   // Here understanding monacoEditor.editor.IStandaloneEditorConstructionOptions
        monaco.editor.create(editorContainer, {
          value: '// some comment',
          language: 'javascript',
        })
```

**Comment by suren-atoyan** [comment]
Actually, there are basic type definitions for the library, please check [this](https://github.com/suren-atoyan/monaco-loader/blob/master/src/types.d.ts)

**Reply by Fi1osof** [reply]
> Actually, there are basic type definitions for the library, please check [this](https://github.com/suren-atoyan/monaco-loader/blob/master/src/types.d.ts)

Yes, i do this.
import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api'

But no types for loader.config({...})
